src/helpers.jl

```diff
@@ -15,11 +20,7 @@ Count the number of columns of a table.
 function count_columns(data)::Integer
     data_schema = Tables.schema(data)
     if isnothing(data_schema)
-        if Tables.columnaccess(data)
-            return length(Tables.columns(data))
-        else
-            @error "Can't count the number of columns"
-        end
+        return length(Tables.columnnames(data))
```

Comment:
Where does `data` come from in this case? Have we already called `Tables.columns` or `Tables.rows`? If we haven't, then calling `Tables.schema`/`Tables.columnnames` aren't going to work in all cases. Maybe we should add some more docs in the Tables.jl package around when certain functions are allowed to be called.

* `Tables.schema` should only be called on the result of `Tables.columns`/`Tables.rows`
* `Tables.columnnames` should only be called on the result of `Tables.columns`, or on each object iterated by the result of `Tables.rows`

As for this function `count_columns`, I'd probably write this like:
```julia
count_columns(data) = length(Tables.Columns(Tables.columns(data)))
```

Reply:
When this function is called `Tables.columns` has already been called, so this should work for now but should probably be refactored in the future, thanks.